### PR TITLE
add documentation on limits and move out Temporal Web docs

### PR DESCRIPTION
**Pull request by sw-yx**
## What does this PR do?

add documentation on limits and move out Temporal Web docs

https://deploy-preview-433--mystifying-fermi-1bc096.netlify.app/docs/server/production-deployment/#server-limits

## Notes to reviewers

the request to document our limits imperfectly as we have them today came from Samar

### CODE REVIEW

File: docs/server/production-deploy.md [hunk 1]
```diff
@@ -44,15 +46,45 @@ You will want to run your own proof of concept tests and watch for key metrics t
...
-Use `workflow_success`, `workflow_failed`, `workflow_timeout`, `workflow_terminate` and `workflow_cancel` counters for each type of Workflow completion.
-They are also include the `namespace` tag.
-Additional information is available in [this forum post](https://community.temporal.io/t/metrics-for-monitoring-server-performance/536/3).
+  These are useful in getting overall stats about Workflow completions.
+  Use `workflow_success`, `workflow_failed`, `workflow_timeout`, `workflow_terminate` and `workflow_cancel` counters for each type of Workflow completion.
+  They are also include the `namespace` tag.
+  Additional information is available in [this forum post](https://community.temporal.io/t/metrics-for-monitoring-server-performance/536/3).
+
+Temporal is highly scalable due to its event sourced design — we have load tested up to 200 million concurrent workflow executions.
+Every shard is low contention by design — it is very difficult to oversubscribe to a task queue in the same cluster.
```
Comment (sw-yx) [inline docs/server/production-deploy.md:60]: if this sounds weird, this directly responds to an answer we gave Instacart https://www.notion.so/temporalio/Instacart-meeting-4-May-2021-004f011d6e054afcbae52ce15af3b0d7

File: docs/system-tools/web-ui.md [hunk 2]
```diff
@@ -4,4 +4,176 @@ title: Temporal Web UI
...
+<summary>
+Optional TLS configuration variables:
+</summary>
+
+| Variable                              | Description                                                         | Default |
+| ------------------------------------- | ------------------------------------------------------------------- | ------- |
+| TEMPORAL_TLS_CERT_PATH                | Certificate for the server to validate the client (web) identity    |         |
+| TEMPORAL_TLS_KEY_PATH                 | Private key for secure communication with the server                |         |
+| TEMPORAL_TLS_CA_PATH                  | Certificate authority (CA) certificate for the validation of server |         |
+| TEMPORAL_TLS_ENABLE_HOST_VERIFICATION | Enables verification of the server certificate                      | true    |
```
Comment (tsurdilo) [inline docs/system-tools/web-ui.md:42]: https://github.com/temporalio/temporal/releases/tag/v1.9.0

this flag was changed :)

File: docs/system-tools/web-ui.md [hunk 3]
```diff
@@ -4,4 +4,176 @@ title: Temporal Web UI
...
+| TEMPORAL_TLS_SERVER_NAME              | Target server that is used for TLS host verification                |         |
+
+- To enable mutual TLS, you need to specify `TEMPORAL_TLS_KEY_PATH` and `TEMPORAL_TLS_CERT_PATH`.
+- For server-side TLS you need to specify only `TEMPORAL_TLS_CA_PATH`.
+
+By default we will also verify your server `hostname`, matching it to `TEMPORAL_TLS_SERVER_NAME`. You can turn this off by setting `TEMPORAL_TLS_ENABLE_HOST_VERIFICATION` to `false`.
```
Comment (tsurdilo) [inline docs/system-tools/web-ui.md:48]: same here

File: docs/system-tools/web-ui.md (hunk 2 above)
Comment (tsurdilo) [reply docs/system-tools/web-ui.md:42]: tls_disable_host_verification (default false) ;)

File: docs/system-tools/web-ui.md (hunk 3 above)
Comment (tsurdilo) [reply docs/system-tools/web-ui.md:48]: tls_disable_host_verification
have to disable it by setting it to true

Comment (sw-yx) [reply docs/system-tools/web-ui.md:48]: not doing this for now bc it doesnt yet apply to temporal web cc @feedmeapples

File: docs/system-tools/web-ui.md [hunk 4]
```diff
@@ -4,4 +4,176 @@ title: Temporal Web UI
...
+
+</details>
+
+### Configuring Authentication (optional)
```
Comment (flossypurse) [inline docs/system-tools/web-ui.md:52]:
```suggestion
### Configuring authentication
```
This causes the url to contain "-optional".
And generally I don't think it is a good practice.
I think it is easy enough to identify this section as optional in one of the first sentences.

File: docs/system-tools/web-ui.md [hunk 5]
```diff
@@ -4,4 +4,176 @@ title: Temporal Web UI
...
+   docker run --network host -v ~/Desktop/config.yml:/usr/app/server/config.yml temporalio/web:latest
+   ```
+
+   </details>
+
+   In future, multiple Oauth providers may be supported, however for now we only read the first Oauth provider under the `providers` key above.
+
+   Common Oauth Providers and their docs:
+
+   - Google: https://developers.google.com/identity/protocols/oauth2/openid-connect
```
Comment (flossypurse) [inline docs/system-tools/web-ui.md:95]: I think these all have bad indents.

File: docs/system-tools/web-ui.md [hunk 6]
```diff
@@ -4,4 +4,176 @@ title: Temporal Web UI
...
+   - please feel free to [PR or request more help on the Temporal Web repo](https://github.com/temporalio/web/)
+
+2. You will need to provide a redirect URL to your Oauth Provider. If you are hosting Temporal Web at `http://localhost:8088` (this is configured by `callback_base_uri` in `server/config.yml`), then it is `http://localhost:8088/auth/sso_callback`.
+
+   - By default, Temporal Web asks for 3 scopes, make sure your provider recognizes these or you may see scope-related errors:
+     - `openid` required by some OIDC providers like [auth0](https://auth0.com/docs/scopes/openid-connect-scopes)
+     - `profile` for name
+     - `email` for email
+
+## Using Temporal Web for Development
```
Comment (flossypurse) [inline docs/system-tools/web-ui.md:126]:
```suggestion
## Using Temporal Web for development
```

File: docs/system-tools/web-ui.md [hunk 7]
```diff
@@ -4,4 +4,176 @@ title: Temporal Web UI
...
+
+Once you have the Temporal Server running locally (use the [quick install guide](/docs/server/quick-install)), you can view the Temporal Web UI at [http://localhost:8088](http://localhost:8088) (this is configurable with the `TEMPORAL_WEB_PORT` environment variable).
+
+> ⚠️ This is a basic guide to troubleshooting/debugging Temporal applications. It is work-in-progress and we encourage [reading about our Architecture](https://docs.temporal.io/docs/server-architecture) for more detail. The better you understand how Temporal works, the better you will be at debugging your workflows.
```
Comment (flossypurse) [inline docs/system-tools/web-ui.md:130]:
```suggestion
> ⚠️ This is a basic guide to troubleshooting/debugging Temporal applications.
> It is work-in-progress and we encourage [reading about our Architecture](https://docs.temporal.io/docs/server-architecture) for more detail.
The better you understand how Temporal works, the better you will be at debugging Workflow Executions.
```

**Review by flossypurse — CHANGES_REQUESTED** [review]

File: docs/server/production-deploy.md [hunk 8]
```diff
@@ -32,6 +32,8 @@ At minimum, the `development.yaml` file needs to have the [`global`](/docs/serve
 
 The [Server configuration reference](/docs/server/configuration) has a more complete list of possible parameters.
 
+Make sure to set Workflow and Activity timeouts everywhere.
```
[the remark this answers is not in the document]
Comment (sw-yx) [reply docs/server/production-deploy.md:35]: idk it was in the original text. i didnt like it either

File: docs/system-tools/web-ui.md [hunk 9]
```diff
@@ -4,4 +4,176 @@ title: Temporal Web UI
...
 ---
 
-Once you have the Temporal Server running locally (use the [quick install guide](/docs/server/quick-install)), you can view the Temporal Web UI at [http://localhost:8088](http://localhost:8088).
+The Temporal Web UI is used to view workflows, see what's running, and explore and debug workflow executions.
+
+For a **video demo** of how this looks, you can [check our docs](https://docs.temporal.io/docs/java-run-your-first-app/#state-visibility).
+
+## Deploying and Securing Temporal Web
+
+_This section is still being written - if you have specific questions you'd like us to answer, please search or [ask on GitHub](https://github.com/temporalio/web/)._
```
[the remark this answers is not in the document]
Comment (sw-yx) [reply docs/system-tools/web-ui.md:13]:
```suggestion
_If you have specific questions you'd like us to answer, please check Slack, Forums, or [post questions on GitHub](https://github.com/temporalio/web/)._
```

File: docs/system-tools/web-ui.md [hunk 10]
```diff
@@ -4,4 +4,176 @@ title: Temporal Web UI
...
+     ```
+
+     </details>
+
+   - Keycloak: https://www.keycloak.org/getting-started/getting-started-docker
+   - please feel free to [PR or request more help on the Temporal Web repo](https://github.com/temporalio/web/)
+
+2. You will need to provide a redirect URL to your Oauth Provider. If you are hosting Temporal Web at `http://localhost:8088` (this is configured by `callback_base_uri` in `server/config.yml`), then it is `http://localhost:8088/auth/sso_callback`.
+
+   - By default, Temporal Web asks for 3 scopes, make sure your provider recognizes these or you may see scope-related errors:
```
[the remark this answers is not in the document]
Comment (sw-yx) [reply docs/system-tools/web-ui.md:125]:
```suggestion
     - By default, Temporal Web asks for 3 scopes, make sure your provider recognizes these or you may see scope-related errors:
```

**Review by flossypurse — APPROVED** [review]

File: docs/server/production-deploy.md [hunk 11]
```diff
@@ -126,10 +116,31 @@ Topics this document will cover in future: (for now, please search/ask on the fo
...
+      - Single payload size
+      - total history size in bytes
+      - single workflow task response size in bytes
+      - Docs Task: [https://www.notion.so/temporalio/Temporal-server-size-limit-doc-352f5720715045aaaba011a94a68718d](https://www.notion.so/temporalio/Temporal-server-size-limit-doc-352f5720715045aaaba011a94a68718d)
+    - a single WF Execution has a critical limit of 50k (todo: CHECK) events - enforced by server -
+      - Dont even get close to 50k - we warn after every 10k events
+      - this is because when worker restarts, we need to replay history
+    - You should write complex workflows if possible... but the scalability unit of our system is a workflow. a single workflow has scalability limitations
+    - Tradeoff: my shared state is lost - i will have to use signals to do IPC
+    - Don't pack everything into a single workflow instance and expect it to scale
```
Comment (sw-yx) [inline docs/server/production-deploy.md:143]:
```suggestion
  - WF as unit of scalability - break things into workflows to scale, don't stuff everything in one workflow!
```

**Pull request merged**